TombEngine/Objects/TR1/Entity/tr1_giant_mutant.cpp

```diff
@@ -187,28 +232,17 @@ namespace TEN::Entities::Creatures::TR1
 
 				break;
 
-			case MUTANT_STATE_ATTACK_3:
+			case MUTANT_STATE_SWEEP_ATTACK:
 				if (item->TouchBits.Test(MutantAttackRightJoints) || LaraItem->HitPoints <= 0)
 				{
-					item->Animation.TargetState = MUTANT_STATE_KILL;
-					Camera.targetDistance = BLOCK(2);
-					Camera.flags = CF_FOLLOW_CENTER;
-
-					SetAnimation(*LaraItem, ID_LARA_EXTRA_ANIMS, LEA_GIANT_MUTANT_DEATH);
-					LaraItem->Animation.IsAirborne = false;
-					LaraItem->Pose = Pose(item->Pose.Position, 0, item->Pose.Orientation.y, 0);
-					LaraItem->RoomNumber = item->RoomNumber;
-					LaraItem->HitPoints = -1;
-					Lara.Status.Air = -1;
-					Lara.Control.HandStatus = HandStatus::Busy;
-					Lara.Control.Weapon.GunType = LaraWeaponType::None;
+					CreatureKill(item, MUTANT_ANIM_KILL, LEA_GIANT_MUTANT_DEATH, MUTANT_STATE_KILL, LS_DEATH);
+					Camera.targetDistance = BLOCK(3);
 				}
 
 				break;
 
 			case MUTANT_STATE_KILL:
-				Camera.targetDistance = BLOCK(2);
```

Comment:
Why these 2 lines were removed here, and why `MaxTurn` was modified instead?

Reply:
Originally, all the enemies had these instructions of the camera and killing animation management, repeated along their code. (Like the enemies who can jump, everyone has the jumping instructions copy/pasted in their control functions).

Tokyo made 1 function and took those repeated lines used by the killing enemies, however, he forgot to include Giant Mutant in there, so this creature still was doing those variations on its own. (Therefore, targetDistance and the CF_FOLLOW_CENTER was being applied in this state).

When I reviewed the killing enemies, I started with the TRex but at the end I went through all of them to ensure that all uses the Tokyo function instead of some using it and other don't. That's why this enemy has so many changed things in comparison with others.

About the MaxTurn, is a fix to prevent the enemy of turning while it's doing the Lara's killing animation. (As Lara position is teleported to the same position than the enemy, many enemies did keep rotating while they do the killing animations).